derive/src/utils.rs

```diff
@@ -127,3 +136,120 @@ pub fn filter_skip_unnamed<'a>(fields: &'a syn::FieldsUnnamed) -> impl Iterator<
 		.enumerate()
 		.filter(|(_, f)| get_skip(&f.attrs).is_none())
 }
+
+pub fn check_attributes(input: &DeriveInput) -> syn::Result<()> {
+	// Check top attribute
+	for attr in &input.attrs {
+		check_top_attribute(attr)?;
+	}
+
+	match input.data {
+		Data::Struct(ref data) => match &data.fields {
+			| Fields::Named(FieldsNamed { named: fields , .. })
+			| Fields::Unnamed(FieldsUnnamed { unnamed: fields, .. }) => {
+				for field in fields {
+					for attr in &field.attrs {
+						check_field_attribute(attr)?;
+					}
+				}
+			}
+			Fields::Unit => (),
+		}
+		Data::Enum(ref data) => {
+			for variant in data.variants.iter() {
+				for attr in &variant.attrs {
+					check_variant_attribute(attr)?;
+				}
+				for field in &variant.fields {
+					for attr in &field.attrs {
+						check_field_attribute(attr)?;
+					}
+				}
+			}
+		},
+		Data::Union(_) => (),
+	}
+	Ok(())
+}
+
+// Is accepted only:
+// * `#[codec(skip)]`
+// * `#[codec(compact)]`
+// * `#[codec(encoded_as = "$EncodeAs")]` with $EncodedAs a valid TokenStream
+fn check_field_attribute(attr: &Attribute) -> syn::Result<()> {
+	let field_error = "Invalid attribute on field, only `#[codec(skip)]`, `#[codec(compact)]` and \
+		`#[codec(encoded_as = \"$EncodeAs\")]` are accepted.";
+
+	if attr.path.is_ident("codec") {
+		match attr.parse_meta()? {
+			Meta::List(ref meta_list) if meta_list.nested.len() == 1 => {
+				match meta_list.nested.first().unwrap() {
+					NestedMeta::Meta(Meta::Path(path))
+						if path.get_ident().map_or(false, |i| i == "skip") => Ok(()),
+
+					NestedMeta::Meta(Meta::Path(path))
+						if path.get_ident().map_or(false, |i| i == "compact") => Ok(()),
+
+					NestedMeta::Meta(Meta::NameValue(MetaNameValue { path, lit: Lit::Str(lit_str), .. }))
+						if path.get_ident().map_or(false, |i| i == "encoded_as")
+					=> TokenStream::from_str(&lit_str.value()).map(|_| ())
+						.map_err(|_e| syn::Error::new(lit_str.span(), "Invalid token stream")),
+
+					elt @ _ => Err(syn::Error::new(elt.span(), field_error)),
+				}
+			},
+			meta @ _ => Err(syn::Error::new(meta.span(), field_error)),
+		}
+	} else {
+		Ok(())
+	}
+}
+
+// Is accepted only:
+// * `#[codec(skip)]`
+// * `#[codec(index = $int)]`
+fn check_variant_attribute(attr: &Attribute) -> syn::Result<()> {
+	let variant_error = "Invalid attribute on variant, only `#[codec(skip)]` and \
+		`#[codec(index = $u8)]` are accepted.";
+
+	if attr.path.is_ident("codec") {
+		match attr.parse_meta()? {
+			Meta::List(ref meta_list) if meta_list.nested.len() == 1 => {
+				match meta_list.nested.first().unwrap() {
+					NestedMeta::Meta(Meta::Path(path))
+						if path.get_ident().map_or(false, |i| i == "skip") => Ok(()),
+
+					NestedMeta::Meta(Meta::NameValue(MetaNameValue { path, lit: Lit::Int(lit_int), .. }))
+						if path.get_ident().map_or(false, |i| i == "index")
+					=> lit_int.base10_parse::<u8>().map(|_| ()),
```

Comment:
here if the error is raised it will output:
```
error: number too large to fit in target type
  --> tests/mod.rs:50:18
   |
50 |     #[codec(index = 15000)]
   |                     ^^^^^

```

Reply:
I think you should generate a custom error here. Saying that the maximum is 256 ;)

Reply:
👍 done